lib/rewards/client/rewards-message-listener.js

```diff
@@ -0,0 +1,116 @@
+const { sc } = require('@reldens/utils');
+const { RewardsConst } = require('../constants');
+const { GameConst } = require('../../game/constants');
+
+class RewardsMessageListener
+{
+
+    static listenMessages(room, gameManager)
+    {
+        room.onMessage('*', (message) => {
+            let rewards = sc.get(message, RewardsConst.REWARDS_CONST, false);
+            if (rewards) {
+                this.loadRewards(rewards, gameManager);
+            }
+
+            if (RewardsConst.REMOVE_DROP === message.act) {
+                this.removeRewardById(message.id, gameManager);
+            }
+        });
+    }
+
+    static loadRewards(rewards, gameManager)
+    {
+        let currentScene = gameManager.getActiveScene();
+        let gameConfig = gameManager.config;
+        let objectPlugin = gameManager.getFeature('objects');
+        let loader = currentScene.load;
+        // TODO: Validate previous required gameManager sections exists.
+
+        for (let [rewardId, reward] of Object.entries(rewards)) {
+            reward['params'] = reward[RewardsConst.REWARDS_PARAMS];
```

Comment:
I got lost with this one, why reasign a value that's already in the reward into another property of the reward?

Reply:
Great question! xD gonna remove it and use the constant instead.